### PR TITLE
MT-22022: Add webhook signature verification helper

### PR DESCRIPTION
**Pull request by Rabsztok**
## Motivation

Expose a helper so Node.js users don't have to re-implement Mailtrap's HMAC-SHA256 webhook signature check on every receiver.

- Algorithm: https://docs.mailtrap.io/email-api-smtp/advanced/webhooks#verifying-the-signature
- Jira: [MT-22022](https://railsware.atlassian.net/browse/MT-22022)

## Changes

- `verifyWebhookSignature(payload, signature, signingSecret)` exported from the package root. HMAC-SHA256 over the raw body, constant-time compare via `crypto.timingSafeEqual`. Accepts `payload` as `string | Buffer`. Returns `false` (never throws) on empty / wrong-length / non-hex / wrong-type inputs.
- `src/__tests__/lib/webhooks/verify-signature.test.ts` pins the cross-SDK fixture (payload + signing_secret + expected digest) shared verbatim across all six official Mailtrap SDKs to guarantee byte-for-byte parity.
- `examples/webhooks/verify-signature.ts` — runnable usage snippet.
- README — new "Verifying webhook signatures" subsection.

## How to test

CI runs jest, eslint, and tsc. Manually:

```sh
yarn ts-node examples/webhooks/verify-signature.ts
```

The script should exit 0 with no output.

## Companion PRs

Coordinated rollout across all six official SDKs (same algorithm, same shared fixture):

- [mailtrap-ruby#110](https://github.com/mailtrap/mailtrap-ruby/pull/110)
- [mailtrap-python#67](https://github.com/mailtrap/mailtrap-python/pull/67)
- [mailtrap-php#66](https://github.com/mailtrap/mailtrap-php/pull/66)
- mailtrap-nodejs#130 (this PR)
- [mailtrap-java#53](https://github.com/mailtrap/mailtrap-java/pull/53)
- [mailtrap-dotnet#242](https://github.com/mailtrap/mailtrap-dotnet/pull/242)

[MT-22022]: https://railsware.atlassian.net/browse/MT-22022?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

## Summary by CodeRabbit

* **New Features**
  * Added a webhook signature verification utility for validating incoming Mailtrap webhooks (HMAC-SHA256).

* **Documentation**
  * README updated with a "Verifying webhook signatures" section and usage guidance.

* **Examples**
  * Added an example HTTP server demonstrating signature verification and proper 200/401 responses.

* **Tests**
  * Added comprehensive tests covering valid, invalid, and edge-case signature scenarios.

[Review Change Stack](https://app.coderabbit.ai/change-stack/mailtrap/mailtrap-nodejs/pull/130?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

**Comment by Rabsztok** [comment]
@coderabbitai review

**Review by IgorDobryn — APPROVED** [review]

File: examples/webhooks/verify-signature.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { createServer, IncomingMessage, ServerResponse } from "http";
+
+// In a real project, this import would be `import { verifyWebhookSignature } from "mailtrap";`
+import { verifyWebhookSignature } from "../../src";
```
Comment (mklocek) [inline examples/webhooks/verify-signature.ts]: Hmm, these examples are for our users. Let's give them the import they would actually use.

```suggestion
import { verifyWebhookSignature } from "mailtrap";
```

**Review by mklocek — APPROVED** [review]
